--- a/clients/terminal.c	2016-05-19 22:36:04.000000000 +0100
+++ b/clients/terminal.c	2016-11-29 18:17:41.532044929 +0000
@@ -53,11 +53,12 @@
 static int option_font_size;
 static char *option_term;
 static char *option_shell;
+static char *option_title;
 
 static struct wl_list terminal_list;
 
 static struct terminal *
-terminal_create(struct display *display);
+terminal_create(struct display *display, const char *title);
 static void
 terminal_destroy(struct terminal *terminal);
 static int
@@ -2272,11 +2273,11 @@
 }
 
 static void
-terminal_new_instance(struct terminal *terminal)
+terminal_new_instance(struct terminal *terminal, const char *title)
 {
 	struct terminal *new_terminal;
 
-	new_terminal = terminal_create(terminal->display);
+	new_terminal = terminal_create(terminal->display, title);
 	if (terminal_run(new_terminal, option_shell))
 		terminal_destroy(new_terminal);
 }
@@ -2296,7 +2297,7 @@
 		terminal_paste(terminal, input);
 		return 1;
 	case XKB_KEY_N:
-		terminal_new_instance(terminal);
+		terminal_new_instance(terminal, option_title);
 		return 1;
 
 	case XKB_KEY_Up:
@@ -2699,7 +2700,7 @@
 
 	switch (index) {
 	case 0:
-		terminal_new_instance(terminal);
+		terminal_new_instance(terminal, option_title);
 		break;
 	case 1:
 		terminal_copy(terminal, input);
@@ -2899,7 +2900,7 @@
 #endif
 
 static struct terminal *
-terminal_create(struct display *display)
+terminal_create(struct display *display, const char *title)
 {
 	struct terminal *terminal;
 	cairo_surface_t *surface;
@@ -2913,7 +2914,7 @@
 	terminal->margin_bottom = -1;
 	terminal->window = window_create(display);
 	terminal->widget = window_frame_create(terminal->window, terminal);
-	terminal->title = xstrdup("Wayland Terminal");
+	terminal->title = xstrdup(title);
 	window_set_title(terminal->window, terminal->title);
 	widget_set_transparent(terminal->widget, 0);
 
@@ -3056,6 +3057,7 @@
 	{ WESTON_OPTION_STRING, "font", 0, &option_font },
 	{ WESTON_OPTION_INTEGER, "font-size", 0, &option_font_size },
 	{ WESTON_OPTION_STRING, "shell", 0, &option_shell },
+	{ WESTON_OPTION_STRING, "title", 0, &option_title },
 };
 
 int main(int argc, char *argv[])
@@ -3080,11 +3082,13 @@
 	weston_config_section_get_string(s, "font", &option_font, "mono");
 	weston_config_section_get_int(s, "font-size", &option_font_size, 14);
 	weston_config_section_get_string(s, "term", &option_term, "xterm");
+	weston_config_section_get_string(s, "title", &option_title, "elephos");
 	weston_config_destroy(config);
 
 	if (parse_options(terminal_options,
 			  ARRAY_LENGTH(terminal_options), &argc, argv) > 1) {
 		printf("Usage: %s [OPTIONS]\n"
+			   "  --title=TITLE\n"
 		       "  --fullscreen or -f\n"
 		       "  --font=NAME\n"
 		       "  --font-size=SIZE\n"
@@ -3099,7 +3103,7 @@
 	}
 
 	wl_list_init(&terminal_list);
-	terminal = terminal_create(d);
+	terminal = terminal_create(d, option_title);
 	if (terminal_run(terminal, option_shell))
 		exit(EXIT_FAILURE);
 
